src/contributing.md

```diff
@@ -0,0 +1,140 @@
+# Contributing guidelines
+
+## Generic Reusable Component Design
+
+Define the component's purpose. Narrow down the scope as much as possible.
+
+Avoid assumptions about its parent/owner.
+
+Define the interface/contract. Make explicit what characteristics can be set by the owner and which are read-only or undefined behavior. (prefix with `_` private internal properties?)
+
+If control maps to a QML control, use QML2's customization mechanisms. [Check docs for examples](https://doc.qt.io/qt-6/qtquickcontrols2-customize.html)
+
+Ensure proper Keyboard support and focus-scope handling.
```

Comment:
I agree that keyboard support is very important and we should have it, but we never did before, so I wonder if maybe we should add some kind of baseline for it before advocating for it (to potential new contributors)

Reply:
I removed it and keep it for a later discussion.